### PR TITLE
fix(docz-core): add support for hoisted dependencies

### LINKED ISSUE #1094: Module not found: Error: Can't resolve 'react' in monorepo.
## Bug Report

2.x releases of docz gives `Module not found` errors when trying to start it with `docz dev`.

**To Reproduce**

1. Clone [this repo](https://github.com/Mefu/docz-error-reproduce)
2. `yarn` to install dependencies.
3. `yarn workspace test-dev-1 start:docz` to reproduce error.
4. `yarn workspace test-dev-2 start:docz` to see it is working correctly with docz 1.x

**Expected behavior**

I believe, I did not miss anything in migration guide and this should work correctly.

**Environment**

* OS: Arch Linux, kernel 5.2.11
* Node/npm version:
   * yarn: 1.17.3
   * node: 12.9.1

**Additional context/Screenshots**
* Project lives in a monorepo, possibly related.
* mdx files import components from files compiled by rollup, possibly unrelated.

### PR DESCRIPTION
**Pull request by rakannimer**
### Description

Before this PR, to use docz in a monorepo you would have to install and use it from the root of the repo only.

This PR allows docz to be used in packages with hoisted node_modules.

- Adds `repoRootPath` config variable to doczrc (defaults to `../`)
- Adds example of monorepo usage without `repoRootPath`
```sh
npx create-docz-app monorepo-docz-in-root --example monorepo
```
- Adds example of monorepo docz usage in package : 
```sh
npx create-docz-app monorepo-docz-in-package --example monorepo-package
```

Fixes #1094 #1014 #933

### CODE REVIEW

**Comment by selbekk** [comment]
😍

I have the following structure in my monorepo (a design system):

```
- packages
  - package-a
  - package-b
  - package-c
- docs
```

I keep it outside of packages, since it's not really a package I'm releasing, but a docs website. It is, however, specified as a workspace in Lerna workspaces.

Will this change affect me at all?

File: examples/monorepo-package/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "private": true,
+  "workspaces": [
+    "packages/*"
+  ],
+  "name": "docz-gatsby-monorepo",
+  "version": "2.0.0-rc.41",
```
Comment (selbekk) [inline examples/monorepo-package/package.json]: This should probably get a version bump?

File: examples/monorepo-package/packages/basic/doczrc.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default {
+  menu: ['Getting Started', 'Components'],
+  repoRootPath: '../..',
```
Comment (selbekk) [inline examples/monorepo-package/packages/basic/doczrc.js]: could we skip this by looking up the tree for a `package.json` instead? If you have something super special, I guess you could set this as a last resort - but stuff that just works is a great deal!

File: tmp.code-workspace [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    {
+      "path": "examples/basic"
+    }
+  ],
+  "settings": {
+    "typescript.tsdk": "node_modules/typescript/lib",
+    "javascript.format.enable": false
+  }
+}
```
Comment (selbekk) [inline tmp.code-workspace]: Should this be added, or ignored?

**Review by selbekk — APPROVED** [review]

File: tmp.code-workspace (hunk 3 above)
Comment (rakannimer) [reply tmp.code-workspace]: Thanks for catching that !

File: examples/monorepo-package/packages/basic/doczrc.js (hunk 2 above)
Comment (rakannimer) [reply examples/monorepo-package/packages/basic/doczrc.js]: Yeah I don't like it either and was really hesitant to add (yet another) config variable, but I couldn't think of a way to find the monorepo root.

> could we skip this by looking up the tree for a package.json instead?

We could, but how would we decide which parent directory to use ? 

For example for the following structure :

```
monorepo/
	package.json
	packages/
		basic/
			package.json
			.docz/
				package.json

```
The `basic` package could have its dependencies in basic/node_modules or monorepo/node_modules depending on the workspace configuration.

Any thoughts on a way around this ?

File: examples/monorepo-package/package.json (hunk 1 above)
Comment (rakannimer) [reply examples/monorepo-package/package.json]: This is the example package, it will never be published to npm, so the version doesn't really matter.

**Comment by rakannimer** [comment]
> I have the following structure in my monorepo (a design system):
> 
> ```
> - packages
>   - package-a
>   - package-b
>   - package-c
> - docs
> ```
> 
> I keep it outside of packages, since it's not really a package I'm releasing, but a docs website. It is, however, specified as a workspace in Lerna workspaces.
> 
> Will this change affect me at all?

Inside docs you're probably using a gatsby app with the gatsby-theme-docz. 

With this change you should be able to use `docz` inside docs like you would in a normal repo by using mdx files only.

You could also use docz from the root of your workspace: 

1 - yarn add docz@next -W
2 - Scatter your mdx files inside package-a, package-b and package-c
3 - yarn docz dev|build|serve

File: examples/monorepo-package/package.json (hunk 1 above)
Comment (selbekk) [reply examples/monorepo-package/package.json]: ah it's the version, not the dependency. My bad 🙈

File: examples/monorepo-package/packages/basic/doczrc.js (hunk 2 above)
Comment (selbekk) [reply examples/monorepo-package/packages/basic/doczrc.js]: 🤔
We could check for the existence of the gatsby binary file - that's the only use we have for this (at least for now, right)? For the project root, we could just find the package.json furthest up the tree.

**Comment by selbekk** [comment]
👏👏👏

File: examples/monorepo-package/packages/basic/doczrc.js (hunk 2 above)
Comment (rakannimer) [reply examples/monorepo-package/packages/basic/doczrc.js]: It now looks for the first gatsby dependency up the tree and uses that .
I left `repoRootPath` in the config as an escape hatch, for people that have very particular setups.

Comment (selbekk) [reply examples/monorepo-package/packages/basic/doczrc.js]: Amazing job 💪 remember to document it!

**Pull request merged**